sys/net/gnrc/network_layer/ipv6/nib/_nib-internal.h

```diff
@@ -0,0 +1,476 @@
+     * @brief   number of unsolicited router advertisements sent
+     *
+     * This only counts up to the first @ref NDP_MAX_INIT_RA_NUMOF on interface
+     * initialization. The last @ref NDP_MAX_FIN_RA_NUMOF of an advertising
+     * interface are counted from UINT8_MAX - @ref NDP_MAX_FIN_RA_NUMOF + 1.
+     *
+     * @note    Only available if @ref GNRC_IPV6_NIB_CONF_ROUTER
+     */
+    uint8_t ra_sent;
+#endif
+    uint8_t rs_sent;                    /**< number of unsolicited router solicitations scheduled */
+    /**
+     * @brief   number of unsolicited neighbor advertisements scheduled
+     */
+    uint8_t na_sent;
+} _nib_iface_t;
+
+/**
+ * @brief   Mutex for locking the NIB.
+ */
+extern mutex_t _nib_mutex;
+
+/**
+ * @brief   Initializes NIB internally
+ */
+void _nib_init(void);
+
+/**
+ * @brief   Gets interface identifier from a NIB entry
+ *
+ * @param[in] nib   A NIB entry
+ *
+ * @return  The NIB entry's interface identifier
+ */
+static inline unsigned _nib_get_if(const _nib_t *nib)
+{
+    return (nib->info & _NIB_IF_MASK) >> _NIB_IF_POS;
+}
+
+/**
+ * @brief   Sets interface for a NIB entry
+ *
+ * @param[in,out] nib   A NIB entry
+ * @param[in] iface     An interface identifier
+ */
+static inline void _nib_set_if(_nib_t *nib, unsigned iface)
+{
+    assert(iface <= 0x1f);
+    nib->info &= ~(_NIB_IF_MASK);
+    nib->info |= ((iface << _NIB_IF_POS) & _NIB_IF_MASK);
+}
+
+/**
+ * @brief   Creates or gets an existing on-link entry by address
+ *
+ * @pre     `(addr != NULL)`
+ *
+ * @param[in] addr  An IPv6 address. May not be NULL.
+ *                  *May also be a global address!*
+ * @param[in] iface The interface to the node.
+ *
+ * @return  A new or existing on-link entry with _nib_t::ipv6 set to @p addr.
+ * @return  NULL, if no space is left.
+ */
+_nib_t *_nib_alloc(const ipv6_addr_t *addr, unsigned iface);
+
+/**
+ * @brief   Clears out a NIB entry (on-link version)
+ *
+ * @param[in] nib  An entry
+ *
+ * @return  true, if entry was cleared
+ * @return  false, if entry was not cleared
+ */
+static inline bool _nib_clear(_nib_t *nib)
+{
+    if (nib->mode == _EMPTY) {
+        memset(nib, 0, sizeof(_nib_t));
+        return true;
+    }
+    return false;
+}
+
+/**
+ * @brief   Iterates over on-link entries
+ *
+ * @param[in] last  Last entry (NULL to start)
+ *
+ * @return  entry after @p last
+ */
+_nib_t *_nib_iter(const _nib_t *last);
+
+/**
+ * @brief   Gets a node by IPv6 address and interface
+ *
+ * @pre     `(addr != NULL)`
+ *
+ * @param[in] addr  The address of a node. May not be NULL.
+ * @param[in] iface The interface to the node. May be 0 for any interface
+ *
+ * @return  The NIB entry for node with @p addr and @p iface on success.
+ * @return  NULL, if there is no such entry.
+ */
+_nib_t *_nib_get(const ipv6_addr_t *addr, unsigned iface);
+
+/**
+ * @brief   Adds a node to the neighbor cache
+ *
+ * @pre     `(addr != NULL)`
+ * @pre     `(cstate != GNRC_IPV6_NIB_NC_INFO_NUD_STATE_DELAY)`
```

Comment:
`@pre ((cstate & GNRC_IPV6_NIB_NC_INFO_NUD_STATE_MASK) != GNRC_IPV6_NIB_NC_INFO_NUD_STATE_DELAY)`

here and below